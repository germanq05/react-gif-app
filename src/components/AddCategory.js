import React, { useState } from 'react'
import PropTypes from 'prop-types';

export const AddCategory = ({ setCategories }) => {

    const [inputValue, setInputValue] = useState('');

    const handleInputChange = (e) => {
        setInputValue(e.target.value)

        console.log('handleInputChange llamado');
    }

    const handleSubmit = (e) => {
        e.preventDefault();
        if (inputValue.trim().length > 2) {
            setCategories(cats => [inputValue, ...cats]);
            setInputValue('');
        }
    }

    return (
        <form onSubmit={handleSubmit}>
            <p> { inputValue } </p>
            <input type="text" placeholder="Agregue categoria"
                value={inputValue} onChange={handleInputChange} />
        </form>
    )
}

AddCategory.protoTypes = {
    setCategories: PropTypes.func.isRequired
}
