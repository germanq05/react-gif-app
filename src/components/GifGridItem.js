import React from 'react';
import PropTypes from 'prop-types';

const GifGridItem = ( {title, url} ) => {

    return (
        <div className="card">
            <img src={ url } alt={ url }/>
            <p>{title}</p>
        </div>
    )
}

export default GifGridItem;

GifGridItem.PropTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired,
}
