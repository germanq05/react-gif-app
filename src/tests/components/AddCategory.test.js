import React from 'react';
import '@testing-library/jest-dom';

import { shallow } from "enzyme"
import { AddCategory } from "../../components/AddCategory"

describe('Pruebas en AddCategory', () => {
    
    const setCategories = jest.fn();
    let wrapper = shallow( <AddCategory setCategories={ setCategories } /> );

    beforeEach( () => {
        jest.clearAllMocks();
        wrapper = shallow( <AddCategory setCategories={ setCategories } /> );
    });

    test('Debe mostrarse correctamente', () => {
        expect( wrapper ).toMatchSnapshot();
    })

    test('Debe de cambiar la caja de texto', () => {
        const input = wrapper.find('input');
        const value = 'Hola Mundo';
        input.simulate('change', {
            target: { value } 
        });

        expect( wrapper.find('p').text().trim() ).toBe( value );
    })
    
    test('No debe de postear la informacion con submit', () => {
        
        wrapper.find('form').simulate('submit', { 
            preventDefault(){} 
        });
        
        expect( setCategories ).not.toHaveBeenCalled(); //No se tiene que haber llamado
    })
    
    test('Debe de llamar el setCategories y limpiar la caja de texto', () => {
        
        const value = 'Hola Mundo'
        
        //Simular el inputChange
        wrapper.find('input').simulate('change', { target: { value } });
        
        //Simular el submit
        wrapper.find('form').simulate('submit', { preventDefault(){} });

        //setCategories se debe haber llamado
        expect( setCategories ).toHaveBeenCalled();
        //Se espera que reciba una funcion
        expect( setCategories ).toHaveBeenCalledWith( expect.any( Function ) );

        //El valor del input debe estar ''
        expect( wrapper.find('input').prop('value') ).toBe('');

    })
    

})
