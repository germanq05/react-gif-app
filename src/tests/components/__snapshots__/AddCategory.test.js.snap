// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en AddCategory Debe mostrarse correctamente 1`] = `
<form
  onSubmit={[Function]}
>
  <p>
     
     
  </p>
  <input
    onChange={[Function]}
    placeholder="Agregue categoria"
    type="text"
    value=""
  />
</form>
`;
